In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib.axes as ax
import numpy as np
import seaborn as sns
import statistics as stas

<h1><center>Estadistica Descriptiva</center></h1>

![](https://images.unsplash.com/flagged/photo-1553267252-d100936057c1?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=470&q=100)

<h1><center>Implementación del DATASET</center></h1>

In [13]:
namesD= {'Piases','Población'}
df = pd.read_csv('population.csv')
df.sample(10)

Unnamed: 0      Value
174             Grenada     111454
169              Belize     383071
51             Cameroon   25216237
56            Sri Lanka   21670000
6               Nigeria  195874740
147              Kosovo    1845300
72             Zimbabwe   14439018
85   Dominican Republic   10627165
117          Costa Rica    4999441
7            Bangladesh  161356039

<h1><center>Medidas de Frecuencia</center></h1>

# Porcentaje

In [49]:
df1 = df.groupby('Unnamed: 0', as_index=False)['Value'].sum()
df1['Percentage'] = df1['Value'] / df1['Value'].sum()
df1 = df1.sort_values('Value', ascending=False)
print (df1)

                Unnamed: 0       Value  Percentage
36                   China  1392730000    0.184502
77                   India  1352617328    0.179188
171                     US   326687501    0.043278
78               Indonesia   267663435    0.035459
127               Pakistan   212215030    0.028113
..                     ...         ...         ...
3                  Andorra       77006    0.000010
49                Dominica       71625    0.000009
139  Saint Kitts and Nevis       52441    0.000007
112                 Monaco       38682    0.000005
99           Liechtenstein       37910    0.000005

[183 rows x 3 columns]


Los datos que podemos visualizar en la parte del porcentaje nos indica la cantidad de ciudadanos que hay con respecto a su ciudad. Es decir se dividio entre la cantidad con respecto a su población para así evaluar cual es el índice de personas en cada país

# Count


In [52]:
count = df.groupby(['Unnamed: 0']).agg({'Value': 'count'}).reset_index()
count

Unnamed: 0  Value
0           Afghanistan      1
1               Albania      1
2               Algeria      1
3               Andorra      1
4                Angola      1
..                  ...    ...
178             Vietnam      1
179  West Bank and Gaza      1
180               Yemen      1
181              Zambia      1
182            Zimbabwe      1

[183 rows x 2 columns]

<h1><center>Medidas de Tendencia Central</center></h1>

# Media

In [57]:
def media(df_final):
    '''Cálculo de la media'''
    med = df_final['Value'].sum() / df_final['Value'].count()
    print("La media de todas las prefecturas corresponde: ")
    return round(med,1)
media(df)

La media de todas las paises corresponde: 


41249226.8

Una vez separada la columna sobre la que se trabajará, misma que corresponde a los valores obtenidos de la población existente de los 183 paises ingresados en el dataset, esto en el año 2018. Para el cálculo de la media, se realiza una función que permite usar docstrings para una explicación sencilla sobre lo que hace la línea de código. Dentro de la función se realiza un cálculo de la media, esto teniendo en cuenta que se tomaron únicamente los valores de población de cada país, siendo esta la columna "Values" sobre la que se aplicará el calculo de la media. Una vez calculada la media, se puede obtener el valor promedio cercano a 41249226.8 existente en el año 2018, esto siendo un cálculo global de la columna "Values".

# Moda

In [37]:
df.Value.mode().sum()

7548608512

Para el cálculo de la moda, se toma inicialmente la columna values que existe dentro del dataset "population_2018", mismo que contiene datos de la población de 183 países. Para el cálculo de la moda se crea una funcion que contendrá un docstring que explica de manera sencilla lo que se realiza en las líneas de código. Una vez creada la función, dentro de la misma se realiza el cálculo de la moda, para esto se usa la función "mode()", que toma el valor que más se repite dentro del dataset. El valor arrojado finalmente es 7548608512, siendo este el que más veces existe dentro de la columna Value de cada país.

# Mediana


In [48]:
def median(l):
    """
      Cálculo de la mediana sobre la tabla Values.
    """
    mediana_resul = df.sort_values('Value',ascending=True)
    half = len(l) // 2

    if not len(l) % 2:
        return (l[half - 1] + l[half]) / 2
    return l[half]

print(median(df['Value']))

9630959


Una vez tomados los valores correspondientes a la columna value, misma que procede del dataset "population_2018" que contiene valores de población de 183 países. Para el cálculo de la mediana, inicialmente se crea una función que contendrá un docstring, esto para que sea una descripción sencilla sobre lo que realizará la línea de código. Para el cálculo de la mediana, se toma los valores en un dato mediana_resul, este siendo ordenado mediante sort, una vez ordenado se toma el valor medio dividiendo los datos a la mitad, para finalmente realizar el cálculo del dataset con su valor medio menos uno, sumado este al valor medio del dataset y dividiendo el resultado de estos sobre dos. Una vez calculada la media, se impirme el resultado, siendo este 9630959, dando a entender que este es el valor medio de la población entre los 183 países.

<h1><center>Medidas de Variación</center></h1>

## Varianza

In [33]:
print("Varianza Resultado:",round(df["Value"].var(ddof=0),1))

Varianza Resultado: 2.178806690174346e+16


El resultado describe la varianza calculada del conjunto de datos de la columna Value. Obtenemos una medida absoluta de la dispersión de los valores de la variable Value respecto de su media. Con la ayuda de las librería y pandas se logró hacer el cálculo usando el método var. En sí es representar la variabilidad de nuestro conjunto de datos respecto de la media aritmética de los mismo

# Desviación Típica

In [30]:
print("Desviación Tipica Resultado: ",df["Value"].std(ddof=0))

Desviación Tipica Resultado:  147607814.50093848


El resultado describe la desviación típica calculada del conjunto de datos de la columna Value. Obtenemos otra medida que ofrece información de la dispersión respecto a la media. Con la ayuda de las librería y pandas se logró hacer el cálculo usando el método std. En sí es representar la desviación típica es la raíz cuadrada de la varianza.

<h1><center>Medidas de Posición</center></h1>

# Quantiles

In [63]:
'''Quantiles'''
print("Min: ", np.quantile(df['Value'],0))
print("Q1: ", np.quantile(df['Value'],0.25))
print("Q2: ", np.quantile(df['Value'],0.50))
print("Q3: ", np.quantile(df['Value'],0.75))
print("Max: ", np.quantile(df['Value'],1))

Min:  37910
Q1:  2743863.0
Q2:  9630959.0
Q3:  29631535.0
Max:  1392730000


Consideranod la estadistica obtenida de los resultados se ha considerado 

In [55]:
'''Quantiles'''
df = pd.DataFrame(df, columns = ['Value']) 
  
df['Quantile_rank'] = pd.qcut(df['Value'], 4, 
                               labels = False)   
print(df)

          Value  Quantile_rank
0    1392730000              3
1    1352617328              3
2     326687501              3
3     267663435              3
4     212215030              3
..          ...            ...
178       77006              0
179       71625              0
180       52441              0
181       38682              0
182       37910              0

[183 rows x 2 columns]
